### PR TITLE
Feature: Integration of One-To-Many Data Structure

### PR DESCRIPTION
**Pull request by georgiyazovaliiski**
**Related issue(s)**:
Closes #NONE

**Summary of the change**:
Integrated `FCOneToManyRelations` and replaced `FCInvertibleMap` with `FCMap`.
Unit tests were also updated. 

**External impacts**:
None.

**Applicable documentation**
- [ ] Javadoc
- [ ] HAPI protobuf comments
- [ ] README

### CODE REVIEW

File: hedera-node/src/main/java/com/hedera/services/context/ServicesContext.java [hunk 1]
```diff
@@ -580,9 +579,12 @@
...
 	private AtomicReference<FCMap<MerkleEntityId, MerkleSchedule>> queryableSchedules;
 	private AtomicReference<FCMap<MerkleBlobMeta, MerkleOptionalBlob>> queryableStorage;
-	private AtomicReference<FCMap<MerkleUniqueTokenId, MerkleUniqueToken>> queryableNfts;
+	private AtomicReference<FCMap<MerkleUniqueTokenId, MerkleUniqueToken>> queryableUniqueTokens;
 	private AtomicReference<FCMap<MerkleEntityAssociation, MerkleTokenRelStatus>> queryableTokenAssociations;
-	private AtomicReference<FCInvertibleHashMap<MerkleUniqueTokenId, MerkleUniqueToken, OwnerIdentifier>> queryableUniqueTokens;
+	private AtomicReference<FCOneToManyRelation<EntityId, MerkleUniqueTokenId>> queryableUniqueTokenAssociations;
+	private AtomicReference<FCOneToManyRelation<EntityId, MerkleUniqueTokenId>> queryableUniqueOwnershipAssociations;
+	private FCOneToManyRelation<EntityId, MerkleUniqueTokenId> uniqueTokenAssociations = new FCOneToManyRelation<>();
+	private FCOneToManyRelation<EntityId, MerkleUniqueTokenId> uniqueOwnershipAssociations = new FCOneToManyRelation<>();
```
Comment (tinker-michaelj) [inline hedera-node/src/main/java/com/hedera/services/context/ServicesContext.java:587]: We need to keep these objects in the `ServicesState` so each fast-copy of the `uniqueTokens` FCM can "carry along" fast-copies of the FCOTMR views that are "frozen" at the moment of the fast-copy.

**Review by tinker-michaelj — COMMENTED** [review]
Overall looks great! Cloned this branch to run CI as `one-to-many-plus-master`, and opened [a PR](https://github.com/hashgraph/hedera-services/pull/1674) against it to stabilize one failing test in CI and address my comment on keeping FCOTMRs in-sync with the `uniqueTokens` FCM.

(**NOTE:** ignore the `plus-master` in the name of that branch, it's a clone of this commit 😁 )

File: hedera-node/src/main/java/com/hedera/services/context/ServicesContext.java (hunk 1 above)
Comment (Daniel-K-Ivanov) [reply hedera-node/src/main/java/com/hedera/services/context/ServicesContext.java]: Resolved from the #1674 PR

**Review by tinker-michaelj — APPROVED** [review]
LGTM

**Pull request merged**